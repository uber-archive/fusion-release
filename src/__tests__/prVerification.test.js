/* eslint-env jest, node */

const afterVerification = require('../afterVerification');

// Mock buildkite agent meta-data calls
jest.mock('shelljs', () => {
  const __commandMock__ = jest.fn();
  return {
    __commandMock__,
    exec: command => {
      if (command.includes('"prerelease"')) {
        return false;
      } else if (command.includes('"release-pr-head-sha"')) {
        return 'TEST_SHA';
      } else if (command.includes('"release-pr-head-repo-full-name"')) {
        return 'TEST_OWNER/TEST_REPO';
      } else if (command.includes('"status"')) {
        return __commandMock__();
      }
    },
  };
});

jest.mock('@octokit/rest', () => {
  const createStatus = jest.fn();
  return () => ({
    authenticate: () => {},
    repos: {
      createStatus,
    },
  });
});

const octokit = require('@octokit/rest');
const {__commandMock__} = require('shelljs');

describe('pull request verification', () => {
  test('success - updates github status', async () => {
    process.env.BUILDKITE_BUILD_URL = 'http://buildkite...';
    __commandMock__.mockReturnValueOnce('success');
    afterVerification();
    expect(octokit().repos.createStatus.mock.calls[0][0]).toMatchSnapshot();
  });

  test('failure - updates github status', async () => {
    process.env.BUILDKITE_BUILD_URL = 'http://buildkite...';
    __commandMock__.mockReturnValueOnce('failure');
    afterVerification();
    expect(octokit().repos.createStatus.mock.calls[1][0]).toMatchSnapshot();
  });
});
