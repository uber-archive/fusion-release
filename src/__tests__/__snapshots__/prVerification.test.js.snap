// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pull request verification failure - updates github status 1`] = `
Object {
  "context": "probot/release-verification",
  "description": "Verification build resulted in success",
  "owner": "TEST_OWNER",
  "repo": "TEST_REPO",
  "sha": "TEST_SHA",
  "state": "success",
  "target_url": "http://buildkite...",
}
`;

exports[`pull request verification success - updates github status 1`] = `
Object {
  "context": "probot/release-verification",
  "description": "Verification build resulted in failure",
  "owner": "TEST_OWNER",
  "repo": "TEST_REPO",
  "sha": "TEST_SHA",
  "state": "failure",
  "target_url": "http://buildkite...",
}
`;
